Index: gfx/src/gtk/nsFontMetricsUtils.cpp
===================================================================
RCS file: /cvsroot/mozilla/gfx/src/gtk/nsFontMetricsUtils.cpp,v
retrieving revision 1.6
diff -d -u -p -r1.6 nsFontMetricsUtils.cpp
--- gfx/src/gtk/nsFontMetricsUtils.cpp	28 Mar 2005 02:14:17 -0000	1.6
+++ gfx/src/gtk/nsFontMetricsUtils.cpp	12 Mar 2006 00:34:40 -0000
@@ -143,11 +143,24 @@ NS_IsXftEnabled(void)
 PRBool
 NS_IsPangoEnabled(void)
 {
-    char *val = PR_GetEnv("MOZ_DISABLE_PANGO");
-    if (val)
-        return FALSE;
+    static PRBool beenHere;
+    static PRBool pangoEnabled;
 
-    return TRUE;
+    if (!beenHere) {
+        beenHere = PR_TRUE;
+
+        char *val = PR_GetEnv("MOZ_DISABLE_PANGO");
+        pangoEnabled = !(val);
+
+        if (pangoEnabled) {
+            nsCOMPtr<nsIPref> prefService = do_GetService(NS_PREF_CONTRACTID);
+            if (prefService)
+                prefService->SetDefaultCharPref("general.useragent.extra.pango",
+                                                "pango-text");
+        }
+    }
+
+    return pangoEnabled;
 }
 
 #endif
