--- mozilla/xpinstall/packager/packages-unix.smp5	2005-12-22 16:29:39.000000000 +0100
+++ mozilla/xpinstall/packager/packages-unix	2006-01-21 09:03:04.000000000 +0100
@@ -45,9 +45,6 @@
 
 [xpcom]
 bin/libmozjs.so
-bin/libnspr4.so
-bin/libplc4.so
-bin/libplds4.so
 bin/cpu/sparcv8plus/libnspr_flt4.so
 bin/libxpcom.so
 bin/libxpcom_core.so
@@ -463,18 +460,13 @@
 bin/components/talkback/*
 
 [psm]
-bin/libnssckbi.so
 bin/components/libpipboot.so
 bin/components/pipboot.xpt
 bin/components/libpipnss.so
 bin/components/pipnss.xpt
 bin/components/libpippki.so
 bin/components/pippki.xpt
-bin/libnss3.so
-bin/libsmime3.so
 bin/libsoftokn3.chk
-bin/libsoftokn3.so
-bin/libssl3.so
 bin/chrome/pipnss.jar
 bin/chrome/pippki.jar
 ; These are required for solaris 32-bit builds
@@ -499,5 +491,17 @@
 bin/components/spellchecker.xpt
 bin/components/myspell/*
 
+[nspr]
+bin/libnspr4.so
+bin/libplc4.so
+bin/libplds4.so
+
+[nss]
+bin/libnssckbi.so
+bin/libnss3.so
+bin/libsmime3.so
+bin/libsoftokn3.so
+bin/libssl3.so
+
 [reporter]
 bin/chrome/reporter.jar
