Index: xpfe/bootstrap/nsAppRunner.cpp
===================================================================
RCS file: /cvsroot/mozilla/xpfe/bootstrap/nsAppRunner.cpp,v
retrieving revision 1.428
diff -d -u -p -r1.428 nsAppRunner.cpp
--- xpfe/bootstrap/nsAppRunner.cpp	20 Dec 2004 21:53:33 -0000	1.428
+++ xpfe/bootstrap/nsAppRunner.cpp	5 Mar 2005 08:39:12 -0000
@@ -1612,6 +1612,11 @@ static PRBool GetWantSplashScreen(int ar
   return dosplash;
 }
 
+#ifdef MOZ_WIDGET_GTK2
+#include "prlink.h"
+typedef void (*_g_set_application_name_fn)(const gchar *application_name);
+#endif
+
 int main(int argc, char* argv[])
 {
   NS_TIMELINE_MARK("enter main");
@@ -1712,6 +1717,19 @@ int main(int argc, char* argv[])
 #endif
   gtk_init(&argc, &argv);
 
+#if defined(MOZ_WIDGET_GTK2)
+  // g_set_application_name () is only defined in glib2.2 and higher.
+  PRLibrary *glib2;
+  _g_set_application_name_fn _g_set_application_name =
+      (_g_set_application_name_fn)PR_FindFunctionSymbolAndLibrary("g_set_application_name", &glib2);
+  if (_g_set_application_name) {
+    _g_set_application_name("seamonkey"); /* MOZ_APP_NAME */
+  }
+  if (glib2) {
+    PR_UnloadLibrary(glib2);
+  }
+#endif
+
   gtk_widget_set_default_visual(gdk_rgb_get_visual());
   gtk_widget_set_default_colormap(gdk_rgb_get_cmap());
 #endif /* MOZ_WIDGET_GTK || MOZ_WIDGET_GTK2 */
