diff -up ./xpinstall/packager/packages-unix.nsprpack ./xpinstall/packager/packages-unix
--- ./xpinstall/packager/packages-unix.nsprpack	2007-10-19 07:27:07.000000000 +0200
+++ ./xpinstall/packager/packages-unix	2007-10-19 07:27:45.000000000 +0200
@@ -42,9 +42,6 @@ bin/isp/SpamPal.sfd
 
 [xpcom]
 bin/libmozjs.so
-bin/libnspr4.so
-bin/libplc4.so
-bin/libplds4.so
 bin/cpu/sparcv8plus/libnspr_flt4.so
 bin/libxpcom.so
 bin/libxpcom_core.so
@@ -55,6 +52,11 @@ bin/components/libjar50.so
 bin/libxpcom_compat.so
 bin/components/libxpcom_compat_c.so
 
+[nspr]
+bin/libnspr4.so
+bin/libplc4.so
+bin/libplds4.so
+
 [browser]
 bin/seamonkey
 bin/seamonkey-bin
@@ -465,13 +467,17 @@ bin/components/qfaservices.xpt
 bin/components/talkback/*
 
 [psm]
-bin/libnssckbi.so
 bin/components/libpipboot.so
 bin/components/pipboot.xpt
 bin/components/libpipnss.so
 bin/components/pipnss.xpt
 bin/components/libpippki.so
 bin/components/pippki.xpt
+bin/chrome/pipnss.jar
+bin/chrome/pippki.jar
+
+[nss]
+bin/libnssckbi.so
 bin/libnss3.so
 bin/libsmime3.so
 bin/libsoftokn3.chk
@@ -489,8 +495,6 @@ bin/libfreebl_64fpu_3.so
 bin/libfreebl_64int_3.chk
 bin/libfreebl_64int_3.so
 bin/libssl3.so
-bin/chrome/pipnss.jar
-bin/chrome/pippki.jar
 
 [inspector]
 bin/components/inspector-cmdline.js
