---
 js/src/jsarray.c |   13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

Index: mozilla/js/src/jsarray.c
===================================================================
--- mozilla.orig/js/src/jsarray.c	2007-10-09 18:05:26.000000000 +0200
+++ mozilla/js/src/jsarray.c	2007-10-09 18:05:46.000000000 +0200
@@ -1891,19 +1891,24 @@
     if (!proto || !InitArrayObject(cx, proto, 0, NULL))
         return NULL;
     return proto;
 }
 
 JSObject *
 js_NewArrayObject(JSContext *cx, jsuint length, jsval *vector)
 {
+    JSTempValueRooter tvr;
     JSObject *obj;
 
     obj = js_NewObject(cx, &js_ArrayClass, NULL, NULL);
     if (!obj)
         return NULL;
-    if (!InitArrayObject(cx, obj, length, vector)) {
-        cx->weakRoots.newborn[GCX_OBJECT] = NULL;
-        return NULL;
-    }
+
+    JS_PUSH_TEMP_ROOT_OBJECT(cx, obj, &tvr);
+    if (!InitArrayObject(cx, obj, length, vector))
+        obj = NULL;
+    JS_POP_TEMP_ROOT(cx, &tvr);
+
+    /* Set/clear newborn root, in case we lost it.  */
+    cx->weakRoots.newborn[GCX_OBJECT] = (JSGCThing *) obj;
     return obj;
 }
