--- mozilla/xpinstall/packager/packages-unix.nsprpack	2007-02-07 04:21:44.000000000 +0100
+++ mozilla/xpinstall/packager/packages-unix	2007-02-07 04:29:55.000000000 +0100
@@ -42,12 +42,8 @@
 bin/defaults/messenger/SpamCatcher.sfd
 bin/defaults/messenger/SpamPal.sfd
 
-
 [xpcom]
 bin/libmozjs.so
-bin/libnspr4.so
-bin/libplc4.so
-bin/libplds4.so
 bin/cpu/sparcv8plus/libnspr_flt4.so
 bin/libxpcom.so
 bin/libxpcom_core.so
@@ -58,6 +54,11 @@
 bin/libxpcom_compat.so
 bin/components/libxpcom_compat_c.so
 
+[nspr]
+bin/libnspr4.so
+bin/libplc4.so
+bin/libplds4.so
+
 [browser]
 bin/seamonkey
 bin/seamonkey-bin
@@ -469,13 +470,17 @@
 bin/components/talkback/*
 
 [psm]
-bin/libnssckbi.so
 bin/components/libpipboot.so
 bin/components/pipboot.xpt
 bin/components/libpipnss.so
 bin/components/pipnss.xpt
 bin/components/libpippki.so
 bin/components/pippki.xpt
+bin/chrome/pipnss.jar
+bin/chrome/pippki.jar
+
+[nss]
+bin/libnssckbi.so
 bin/libnss3.so
 bin/libsmime3.so
 bin/libsoftokn3.chk
@@ -493,8 +498,6 @@
 bin/libfreebl_64int_3.chk
 bin/libfreebl_64int_3.so
 bin/libssl3.so
-bin/chrome/pipnss.jar
-bin/chrome/pippki.jar
 
 [inspector]
 bin/components/inspector-cmdline.js
