diff -up seamonkey-2.0/comm-central/suite/app/Makefile.in.old seamonkey-2.0/comm-central/suite/app/Makefile.in
--- seamonkey-2.0/comm-central/suite/app/Makefile.in.old	2009-09-03 19:45:03.000000000 +0200
+++ seamonkey-2.0/comm-central/suite/app/Makefile.in	2009-09-23 14:59:11.000000000 +0200
@@ -133,6 +133,9 @@ ifdef MOZ_ENABLE_DBUS
 LIBS += $(MOZ_DBUS_GLIB_LIBS)
 endif
 
+ifdef MOZ_NATIVE_HUNSPELL
+LIBS += $(MOZ_HUNSPELL_LIBS)
+endif
 
 ifndef MOZ_WINCONSOLE
 ifdef MOZ_DEBUG
