Index: mozilla/configure.in
===================================================================
RCS file: /cvsroot/mozilla/configure.in,v
retrieving revision 1.1503.2.15.2.3
diff -u -r1.1503.2.15.2.3 configure.in
--- mozilla/configure.in	10 Jan 2006 03:11:42 -0000	1.1503.2.15.2.3
+++ mozilla/configure.in	21 Jan 2006 09:00:24 -0000
@@ -2383,7 +2383,7 @@
 dnl Check for .hidden assembler directive and visibility attribute.
 dnl Borrowed from glibc configure.in
 dnl ===============================================================
-if test "$GNU_CC"; then
+if /bin/false; then
   AC_CACHE_CHECK(for visibility(hidden) attribute,
                  ac_cv_visibility_hidden,
                  [cat > conftest.c <<EOF
